src/solver/conv_winoRxS_f3x2.cpp

```diff
@@ -32,154 +32,42 @@
-    // If stride = 2: S is always padded to smallest 6*n for some integer n
-    int padded_S = 0;
-    if(S_stride == 1)
-    {
-        if(S <= TILE)
-        {
-            padded_S = TILE;
-        }
-        else
-        {
-            padded_S = Ceiling(S, TILE_X2);
-        }
-    }
-    else
-    {
-        padded_S = Ceiling(S, TILE_X2);
-    }
-    // If stride = 1: R is always padded to smallest 3*m for some integer m
-    // If stride = 2: if R % 6 ==1 then R is padded to smallest 3*m for some
-    // integer m, otherwise R is padded to smallest 6*m for some integer m
-    int padded_R = 0;
-    if(R_stride == 1)
-    {
-        padded_R = Ceiling(R, TILE);
-    }
-    else
-    {
-        if(R % TILE_X2 == 1)
-        {
-            padded_R = Ceiling(R, TILE);
-        }
-        else
-        {
-            padded_R = Ceiling(R, TILE_X2);
-        }
-    }
-    // Check C restrictions:
-    // For FP16, all C restrictions shall be multipled by 2.
-    // This implicitly introduces restriction that C must be even.
-    if(fp16 && C % 2 != 0)
-    {
-        return false;
-    }
-    // If stride == 1 and S <= 3 then C needs to be even, otherwise not
-    if(S_stride == 1 && S <= TILE && C % (fp16 ? 4 : 2) != 0)
-    {
-        return false;
-    }
-    const bool is_dilated_stride_2 = (params.direction.IsBackwardData() && S_stride != 1);
-    if(fp16)
-    {
-        if(is_dilated_stride_2)
-        {
-            if(C % 4 != 0)
-                return false;
-            // In dilation mode with stride== 2 the following should be satisfied:
-            // C * (ceil(R/6) + floor((R+4)/6)) * ceil(S/6) >= 18*2 (fp16)
-            const auto k = CeilDiv(R, TILE_X2) + FloorDiv((R + TILE + 1), TILE_X2);
-            const auto l = CeilDiv(S, TILE_X2);
-            if(C * k * l < 18 * 2)
-                return false;
-        }
-        if(padded_R * padded_S * C < TILE * TILE * 18 * 2)
-            return false;
-    }
-    else
-    {
-        // 9_0_14 readme: Additional limitations in the dilated case are R> 1 and  C %2==0
-        if(is_dilated_stride_2)
-        {
-            if(!(R > 1))
-                return false;
-            if(!(C % 2 == 0))
-                return false;
-        }
-        // If the padded_R x padded_S filter size from above is 3*k x 3*l
-        // or (special case for dilated with stride 2) 3*k x 6*l, then
-        // it should be that k*l*C  >=18
-        assert(padded_R % TILE == 0 && padded_S % (is_dilated_stride_2 ? TILE_X2 : TILE) == 0);
-        const int k = padded_R / TILE;
-        const int l = padded_S / (is_dilated_stride_2 ? TILE_X2 : TILE);
-        if(k * l * C < 18)
-            return false;
-    }
     // Padding for bwd data shall not be negative.
     /// \todo Either remove WrW related code or re-use function from RxS
-    if(params.direction.IsBackwardData() || params.direction.IsBackwardWrW())
+    if(params.direction.IsBackwardData())
```

Comment:
Bugfix?

Reply:
This is an incorrect limitation that originally comes from `conv_winoRxS` solver.
Now wino_f3x2 has the same constraints as `conv_winoRxS_f2x3` solver, where this limitation was  [removed](https://github.com/ROCmSoftwarePlatform/MIOpen/pull/640/commits/c20d531b086b179c5625ec1a642d1505991a8232).